src/background/uri-info.js

```diff
@@ -1,25 +1,72 @@
 import settings from './settings';
 
+const BLOCKLIST = [
+  { urlProperty: 'protocol', value: 'chrome:' },
+  { urlProperty: 'hostname', value: 'facebook.com' },
+  { urlProperty: 'hostname', value: 'www.facebook.com' },
+  { urlProperty: 'hostname', value: 'mail.google.com' },
+];
+
 /** encodeUriQuery encodes a string for use in a query parameter */
 function encodeUriQuery(val) {
   return encodeURIComponent(val).replace(/%20/g, '+');
 }
 
+/**
+ * Should we send a "badge" request to obtain the annotation count for the
+ * URL `uri`?
+ *
+ * @param {string} uri
+ * @return {boolean}
+ */
+function shouldQueryUri(uri) {
+  let url;
+
+  try {
+    url = new URL(uri);
+  } catch (e) {
+    return false;
+  }
+
+  // Make sure `uri` does not match ANY item in the blocklist
+  return BLOCKLIST.every(
+    blockedItem => url[blockedItem.urlProperty] !== blockedItem.value
+  );
+}
+
 /**
  * Queries the Hypothesis service that provides
  * statistics about the annotations for a given URL.
  */
 export function query(uri) {
   return fetch(settings.apiUrl + '/badge?uri=' + encodeUriQuery(uri), {
     credentials: 'include',
-  })
-    .then(function (res) {
-      return res.json();
-    })
-    .then(function (data) {
-      if (typeof data.total !== 'number') {
-        throw new Error('Annotation count is not a number');
-      }
-      return data;
-    });
+  }).then(res => res.json());
+}
+
+/**
+ * Retrieve the count of available annotations for `uri`
+ *
+ * @return {Promise<number>} - Annotation count for `uri`. Will be 0 if URI
+ *                             has a blocklist match or there are problems with
+ *                             the request to or response from the API
+ */
+export function getAnnotationCount(uri) {
+  const noValue = 0;
+  if (shouldQueryUri(uri)) {
+    return query(uri)
+      .then(data => {
+        if (!data || typeof data.total !== 'number') {
+          console.warn('Annotation count is not a number');
```

Comment:
Previously, this code threw on data it didn't like, but from a purist's perspective, the `fetch` operation is successful, so I'd prefer not to throw. The net result was that a `0` value got set for the annotation's count, anyway. And still is.